### PR TITLE
TPD-846 Widget Recent Attendance Trends, and TPD-1020 Widget Giving Metrics, and TPD-1021 Widget Giving Type Breakdown

### PR DESCRIPTION
**Pull request by jason-rice**
the widget works, now it's time to see if the migration works!
[touchpointsoftware.atlassian.net/browse/TPD-846](https://touchpointsoftware.atlassian.net/browse/TPD-846)
[touchpointsoftware.atlassian.net/browse/TPD-1020](https://touchpointsoftware.atlassian.net/browse/TPD-1020)
[touchpointsoftware.atlassian.net/browse/TPD-1021](https://touchpointsoftware.atlassian.net/browse/TPD-1021)

### CODE REVIEW

File: CmsData/Migrations/20200511_add_recent_attendance_trends_widget.sql [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+           ([Id],[Word])
+     VALUES
+           ((select SCOPE_IDENTITY())
+           ,'widget')
+
+INSERT INTO [dbo].[Content]
+           ([Name],[Title],[Body],[DateCreated],[TypeID],[ThumbID],[RoleID],[OwnerID],[CreatedBy])
+     VALUES
+           ('WidgetRecentAttendanceTrendsPython','Edit Python Script',
+           'divisions = [
```
Comment (timdavila) [inline CmsData/Migrations/20200511_add_recent_attendance_trends_widget.sql:74]: I would add a comment here explaining what to put into the array

File: CmsData/Migrations/20200511_add_recent_attendance_trends_widget.sql [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    [''Young Couples'', 210],
+    [''Young Marrieds'', 211],
+    [''Adults 1'', 213]
+]
+begindate = ''11/15/2019''
+beginhour = 1
+endhour = 12
+Title = ''Recent Attendance Trends''
```
Comment (timdavila) [inline CmsData/Migrations/20200511_add_recent_attendance_trends_widget.sql:82]: Same thing here with the begin/end hour. Also is the title necessary? Widget title is already a field the user can set.

File: CmsData/Migrations/20200511_add_recent_attendance_trends_widget.sql [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+           ,0
+           ,(select isnull(max([Order]), 0)+1 from DashboardWidgets)
+           ,1
+           ,1
+           ,6)
+
+INSERT INTO [dbo].[DashboardWidgetRoles]
+           ([WidgetId]
+           ,[RoleId])
+    SELECT SCOPE_IDENTITY() [WidgetId], RoleId FROM dbo.Roles WHERE RoleName in ('Finance', 'FinanceAdmin', 'Admin')
```
Comment (timdavila) [inline CmsData/Migrations/20200511_add_recent_attendance_trends_widget.sql:193]: This seems too strict. Attendance isn't really financial data

File: CmsData/Migrations/20200511_add_recent_attendance_trends_widget.sql [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+    sql = sql.replace(''@beginhour'', str(beginhour))
+    sql = sql.replace(''@endhour'', str(endhour))
+    sql = sql.replace(''@divs'', str(divids)[1:-1])
+    rowcountsql = ''isnull((select sum(HeadCount) from data d where d.DivisionId = {0} and d.ss = dd.ss group by d.ss), 0) d{0}''
+    rlist = [rowcountsql.format(d) for d in divids]
+    s = '',\n''.join(rlist)
+    sql = sql.replace(''@rowcounts'', s)
+    return q.QuerySqlJsonArray(sql)
+
+def GetJavascript(divisions, data, Title):
```
Comment (timdavila) [inline CmsData/Migrations/20200511_add_recent_attendance_trends_widget.sql:98]: This method never seems to be called. Is it needed? Handlebars should be able to handle any text replacement.

File: CmsData/Migrations/20200511_add_recent_attendance_trends_widget.sql (hunk 1 above)
Comment (jason-rice) [reply CmsData/Migrations/20200511_add_recent_attendance_trends_widget.sql:74]: fixed this

File: CmsData/Migrations/20200511_add_recent_attendance_trends_widget.sql (hunk 2 above)
Comment (jason-rice) [reply CmsData/Migrations/20200511_add_recent_attendance_trends_widget.sql:82]: I removed beginhour, endhour and title. These weren't necessary

File: CmsData/Migrations/20200511_add_recent_attendance_trends_widget.sql (hunk 4 above)
Comment (jason-rice) [reply CmsData/Migrations/20200511_add_recent_attendance_trends_widget.sql:98]: Great suggestion! I removed this.

File: CmsData/Migrations/20200511_add_recent_attendance_trends_widget.sql (hunk 3 above)
Comment (jason-rice) [reply CmsData/Migrations/20200511_add_recent_attendance_trends_widget.sql:193]: I changed this to edit only, per instructions from Chris. Good catch Tim!

**Pull request merged**